File: 2. Even Fibonacci Numbers/solution.go
package main

import "fmt"

func fibonacciSum(limit int) int {
	fibSum := 0
	a, b := 1, 2

	for a <= limit {
		if a%2 == 0 {
			fibSum += a
		}
		a, b = b, a+b
	}

	return fibSum
}

func main() {
	limit := 4000000
	result := fibonacciSum(limit)
	fmt.Println(result)
}

File: 2. Even Fibonacci Numbers/solution.java
public class solution {
    public static void main(String[] args) {
        int limit = 4000000;
        int result = fibonacciSum(limit);
        System.out.println(result);
    }

    public static int fibonacciSum(int limit) {
        int fibSum = 0;
        int a = 1;
        int b = 2;

        while (a <= limit) {
            if (a % 2 == 0) {
                fibSum += a;
            }
            int temp = a;
            a = b;
            b = temp + b;
        }

        return fibSum;
    }
}
